tsdb/head_test.go

```diff
@@ -387,11 +387,22 @@ func TestHead_Truncate(t *testing.T) {
 		"2": {},
 	}, h.symbols)
 
+	values := map[string]stringset{}
+	err := h.postings.Iter(func(l labels.Label, _ index.Postings) error {
+		ss, ok := values[l.Name]
+		if !ok {
+			ss = stringset{}
+			values[l.Name] = ss
+		}
+		ss.set(l.Value)
+		return nil
+	})
+	testutil.Equals(t, nil, err)
```

Comment:
How about we use the newly implemented `LabelNames` and `LabelValues` methods for `MemPostings` to build the set? It would also test those methods.